core/src/services/webhdfs/writer.rs

```diff
@@ -17,23 +17,34 @@
 
 use async_trait::async_trait;
 use http::StatusCode;
+use tokio::sync::RwLock;
 
 use super::backend::WebhdfsBackend;
 use super::error::parse_error;
 use crate::raw::oio::WriteBuf;
 use crate::raw::*;
 use crate::*;
 
+pub type WebhdfsWriters =
+    TwoWays<oio::OneShotWriter<WebhdfsWriter>, oio::AppendObjectWriter<WebhdfsWriter>>;
+
 pub struct WebhdfsWriter {
     backend: WebhdfsBackend,
 
     op: OpWrite,
     path: String,
+    /// for `append`
+    location: RwLock<String>,
```

Comment:
It's better to call `op=APPEND` everytime to get the location instead of store it inside writer.

Reply:
Do you mean that offset does nothing and calls op=APPEND in append instead?

Reply:
> Do you mean that offset does nothing and calls op=APPEND in append instead?

Yes, WebHDFS does not require an append-offset to perform an append operation. Perhaps we could consider creating a new type of writer, but it's probably best to leave it as is.

Reply:
fixed